src/App.js

```diff
@@ -7,66 +7,105 @@ import {
   NavLink,
   Redirect,
 } from 'react-router-dom';
+import { db } from './lib/firebase';
+// import { useCollection } from 'react-firebase-hooks/firestore';
+import { SnackbarProvider } from 'notistack';
 
 import AddItems from './AddItems';
 import ItemList from './ItemList';
 import Welcome from './Welcome';
 
-
 function App() {
-  const hasToken = localStorage.getItem('userToken');
+  const [token, setToken] = useState(localStorage.getItem('userToken'));
+  const [list, setList] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    let unsubscribe;
+    try {
+      unsubscribe = db.collection(token).onSnapshot((snapshot) => {
+        const newList = [];
+        snapshot.forEach((doc) => {
+          // This seems to work, while snapshot.map doesn't
+          newList.push(
+            JSON.stringify(doc.data()['formData']['itemName']).replace(
+              /['"]+/g,
+              '',
+            ),
+          );
+        });
+        setList(newList);
+        setError(null);
+        setLoading(false);
+      });
+    } catch (error) {
+      setError("Can't connect to the database");
+      setLoading(false);
+    }
+    return () => {
+      unsubscribe();
+    };
+  }, [token]);
+
```

Comment:
Can't wait to hear about all of the trouble-shooting you did with this. Well done!